fakeredis.aioredis.FakeRedis to support version selection

**Is your feature request related to a problem? Please describe.**

I'm still using Redis 6, so it would be useful to run tests in Redis 6 compatibility mode. This is supported for the blocking client, but doesn't work for the asyncio client:

```
>       async with fakeredis.aioredis.FakeRedis(decode_responses=True, version=6) as client:

tests\test_redis_client.py:33:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

...SNIPPED...

>       super().__init__(
            db=db,
            password=password,
            socket_timeout=socket_timeout,
            connection_pool=connection_pool,
            encoding=encoding,
            encoding_errors=encoding_errors,
            decode_responses=decode_responses,
            retry_on_timeout=retry_on_timeout,
            max_connections=max_connections,
            health_check_interval=health_check_interval,
            client_name=client_name,
            username=username,
            **kwargs,
        )
E       TypeError: Redis.__init__() got an unexpected keyword argument 'version'

\Lib\site-packages\fakeredis\aioredis.py:230: TypeError
```

**Describe the solution you'd like**

fakeredis.aioredis.FakeRedis to support the same `version` parameter as fakeredis.FakeRedis

**Describe alternatives you've considered**

redislite, testing against real redis instance. These are sub-optimal for the same reasons fakeredis in general is good.

--- a/docs/about/changelog.md
+++ b/docs/about/changelog.md
@@ -11,6 +11,10 @@ description: Change log of all fakeredis releases
 
 - Implemented support for `JSON.MSET` #174, `JSON.MERGE` #181
 
+### 🧰 Bug Fixes
+
+- Add support for version for async FakeRedis #205
+
 ### 🧰 Maintenance
 
 - Updated how to test django_rq #204

--- a/docs/redis-commands/Redis.md
+++ b/docs/redis-commands/Redis.md
@@ -1,6 +1,6 @@
 # Redis commands
 
-## server commands
+## `server` commands (8/70 implemented)
 
 ### [BGSAVE](https://redis.io/commands/bgsave/)
 
@@ -287,7 +287,7 @@ Clears all entries from the slow log.
 An internal command used in replication.
 
 
-## string commands
+## `string` commands (22/22 implemented)
 
 ### [APPEND](https://redis.io/commands/append/)
 
@@ -512,7 +512,7 @@ Enables read-only queries for a connection to a Redis Cluster replica node.
 Enables read-write queries for a connection to a Reids Cluster replica node.
 
 
-## connection commands
+## `connection` commands (3/25 implemented)
 
 ### [ECHO](https://redis.io/commands/echo/)
 
@@ -619,7 +619,7 @@ Closes the connection.
 Resets the connection.
 
 
-## bitmap commands
+## `bitmap` commands (5/7 implemented)
 
 ### [BITCOUNT](https://redis.io/commands/bitcount/)
 
@@ -654,7 +654,7 @@ Performs arbitrary bitfield integer operations on strings.
 Performs arbitrary read-only bitfield integer operations on strings.
 
 
-## list commands
+## `list` commands (19/22 implemented)
 
 ### [BLMOVE](https://redis.io/commands/blmove/)
 
@@ -749,7 +749,7 @@ Returns multiple elements from a list after removing them. Deletes the list if t
 Returns the index of matching elements in a list.
 
 
-## sorted-set commands
+## `sorted-set` commands (33/35 implemented)
 
 ### [BZPOPMAX](https://redis.io/commands/bzpopmax/)
 
@@ -896,7 +896,7 @@ Removes and returns a member by score from one or more sorted sets. Blocks until
 Returns the highest- or lowest-scoring members from one or more sorted sets after removing them. Deletes the sorted set if the last member was popped.
 
 
-## generic commands
+## `generic` commands (20/33 implemented)
 
 ### [DEL](https://redis.io/commands/del/)
 
@@ -1035,7 +1035,7 @@ Blocks until the asynchronous replication of all preceding write commands sent b
 Blocks until all of the preceding write commands sent by the connection are written to the append-only file of the master and/or replicas.
 
 
-## transactions commands
+## `transactions` commands (5/5 implemented)
 
 ### [DISCARD](https://redis.io/commands/discard/)
 
@@ -1059,7 +1059,7 @@ Monitors changes to keys to determine the execution of a transaction.
 
 
 
-## scripting commands
+## `scripting` commands (7/22 implemented)
 
 ### [EVAL](https://redis.io/commands/eval/)
 
@@ -1154,7 +1154,7 @@ Sets the debug mode of server-side Lua scripts.
 Terminates a server-side Lua script during execution.
 
 
-## geo commands
+## `geo` commands (10/10 implemented)
 
 ### [GEOADD](https://redis.io/commands/geoadd/)
 
@@ -1198,7 +1198,7 @@ Queries a geospatial index for members inside an area of a box or a circle, opti
 
 
 
-## hash commands
+## `hash` commands (16/16 implemented)
 
 ### [HDEL](https://redis.io/commands/hdel/)
 
@@ -1266,7 +1266,7 @@ Returns all values in a hash.
 
 
 
-## hyperloglog commands
+## `hyperloglog` commands (3/5 implemented)
 
 ### [PFADD](https://redis.io/commands/pfadd/)
 
@@ -1293,7 +1293,7 @@ Internal commands for debugging HyperLogLog values.
 An internal command for testing HyperLogLog values.
 
 
-## pubsub commands
+## `pubsub` commands (9/15 implemented)
 
 ### [PSUBSCRIBE](https://redis.io/commands/psubscribe/)
 
@@ -1360,7 +1360,7 @@ Listens for messages published to shard channels.
 Stops listening to messages posted to shard channels.
 
 
-## set commands
+## `set` commands (17/17 implemented)
 
 ### [SADD](https://redis.io/commands/sadd/)
 
@@ -1432,7 +1432,7 @@ Stores the union of multiple sets in a key.
 
 
 
-## stream commands
+## `stream` commands (20/20 implemented)
 
 ### [XACK](https://redis.io/commands/xack/)
 

--- a/docs/redis-commands/RedisJson.md
+++ b/docs/redis-commands/RedisJson.md
@@ -1,6 +1,6 @@
 # JSON commands
 
-## json commands
+## `json` commands (22/22 implemented)
 
 ### [JSON.DEL](https://redis.io/commands/json.del/)
 

--- a/fakeredis/aioredis.py
+++ b/fakeredis/aioredis.py
@@ -8,11 +8,6 @@ import redis
 
 from ._server import FakeBaseConnectionMixin
 
-if sys.version_info >= (3, 8):
-    from typing import Type, TypedDict
-else:
-    from typing_extensions import Type, TypedDict
-
 if sys.version_info >= (3, 11):
     from asyncio import timeout as async_timeout
 else:
@@ -171,22 +166,6 @@ class FakeConnection(FakeBaseConnectionMixin, redis_async.Connection):
         return pieces
 
 
-class ConnectionKwargs(TypedDict, total=False):
-    db: Union[str, int]
-    username: Optional[str]
-    password: Optional[str]
-    socket_timeout: Optional[float]
-    encoding: str
-    encoding_errors: str
-    decode_responses: bool
-    retry_on_timeout: bool
-    health_check_interval: int
-    client_name: Optional[str]
-    server: Optional[_server.FakeServer]
-    connection_class: Type[redis_async.Connection]
-    max_connections: Optional[int]
-
-
 class FakeRedis(redis_async.Redis):
     def __init__(
             self,
@@ -205,27 +184,29 @@ class FakeRedis(redis_async.Redis):
             username: Optional[str] = None,
             server: Optional[_server.FakeServer] = None,
             connected: bool = True,
+            version=(7,),
             **kwargs,
     ):
         if not connection_pool:
             # Adapted from aioredis
-            connection_kwargs: ConnectionKwargs = {
-                "db": db,
+            connection_kwargs = dict(
+                db=db,
                 # Ignoring because AUTH is not implemented
                 # 'username',
                 # 'password',
-                "socket_timeout": socket_timeout,
-                "encoding": encoding,
-                "encoding_errors": encoding_errors,
-                "decode_responses": decode_responses,
-                "retry_on_timeout": retry_on_timeout,
-                "health_check_interval": health_check_interval,
-                "client_name": client_name,
-                "server": server,
-                "connected": connected,
-                "connection_class": FakeConnection,
-                "max_connections": max_connections,
-            }
+                socket_timeout=socket_timeout,
+                encoding=encoding,
+                encoding_errors=encoding_errors,
+                decode_responses=decode_responses,
+                retry_on_timeout=retry_on_timeout,
+                health_check_interval=health_check_interval,
+                client_name=client_name,
+                server=server,
+                connected=connected,
+                connection_class=FakeConnection,
+                max_connections=max_connections,
+                version=version,
+            )
             connection_pool = redis_async.ConnectionPool(**connection_kwargs)
         super().__init__(
             db=db,

--- a/scripts/supported2.py
+++ b/scripts/supported2.py
@@ -63,14 +63,17 @@ def generate_markdown_files(commands: dict, implemented_commands: set[str], stac
         f.write('Module currently not implemented in fakeredis.\n\n')
     for group in groups:
         implemented_in_group = list(filter(lambda cmd: cmd in implemented_commands, groups[group]))
+        unimplemented_in_group = list(filter(
+            lambda cmd: cmd not in implemented_commands and cmd.upper() not in IGNORE_COMMANDS, groups[group]))
         if len(implemented_in_group) > 0:
-            f.write(f'## {group} commands\n\n')
+            f.write(f'## `{group}` commands '
+                    f'({len(implemented_in_group)}/{len(unimplemented_in_group) + len(implemented_in_group)} '
+                    f'implemented)\n\n')
         for cmd in implemented_in_group:
             f.write(f"### [{cmd.upper()}](https://redis.io/commands/{cmd.replace(' ', '-')}/)\n\n")
             f.write(f"{commands[cmd]['summary']}\n\n")
         f.write("\n")
-        unimplemented_in_group = list(filter(
-            lambda cmd: cmd not in implemented_commands and cmd.upper() not in IGNORE_COMMANDS, groups[group]))
+
         if len(unimplemented_in_group) > 0:
             f.write(f'### Unsupported {group} commands \n')
             f.write('> To implement support for a command, see [here](../../guides/implement-command/) \n\n')

--- a/test/test_redis_asyncio.py
+++ b/test/test_redis_asyncio.py
@@ -274,6 +274,19 @@ async def test_from_url():
     await r1.connection_pool.disconnect()
 
 
+@pytest.mark.fake
+async def test_from_url_with_version():
+    r0 = aioredis.FakeRedis.from_url('redis://localhost?db=0', version=(6,))
+    r1 = aioredis.FakeRedis.from_url('redis://localhost?db=1', version=(6,))
+    # Check that they are indeed different databases
+    await r0.set('foo', 'a')
+    await r1.set('foo', 'b')
+    assert await r0.get('foo') == b'a'
+    assert await r1.get('foo') == b'b'
+    await r0.connection_pool.disconnect()
+    await r1.connection_pool.disconnect()
+
+
 @fake_only
 async def test_from_url_with_server(req_aioredis2, fake_server):
     r2 = aioredis.FakeRedis.from_url('redis://localhost', server=fake_server)

